src/keystore.h

```diff
@@ -19,17 +19,24 @@
 /** A virtual base class for key stores */
 class CKeyStore
 {
-protected:
+public:
+    /** This lock only needs to be explicitly taken by the caller if lock-free functions are called.  All lock-free
+        functions are preceded by an _.  In general, use the locking functions.  Only use the lock-free functions
+        for optimization of loops by factoring the lock out of the loop and calling the lock-free function in the loop.
+    */
     mutable CCriticalSection cs_KeyStore;
 
-public:
```

Comment:
why made `public` also all the following, wasn't `protected` enough?

Reply:
I just need cs_KeyStore to be public, for optimization reasons.  I factored the lock/unlock out of an inner loop and added a few non-locking _ functions to this class.

Reply:
I see. I misread the patch all the virtual functions members and deconstructor were already public. sorry for the noise